gazebosim/gz-tools
# Use gz-cmake

# 🎉 New feature

Requires
* https://github.com/osrf/homebrew-simulation/pull/1944
* https://github.com/gazebo-tooling/release-tools/pull/749
* https://github.com/gazebo-tooling/gazebodistro/pull/113

## Summary


Reduce duplication with other Gazebo libraries by depending on `gz-cmake`.

There's been a previous attempt at migrating to `gz-cmake` in [PR 6](https://osrf-migration.github.io/ignition-gh-pages/#!/ignitionrobotics/ign-tools/pull-requests/6/page/1), but at the time we decided against it because it would be complicated to support two versions installing the `ign` executable. With the move to `gz` (see https://github.com/gazebo-tooling/release-tools/issues/739), that's no longer an issue.

## Test it


* Build this package with colcon
* Build some package that depends on it, like `gz-plugin`, and check that the `gz plugin` CLI works
* Run `colcon test` and check that `UNIT_gz_TEST` and `rubocop` passed
* Run `colcon build --cmake-target doc`, then navigate to `build/ignition-tools2/doxygen/html/` and check that the documentation site is correctly generated
* Run `colcon build --cmake-target codecheck` and check it passes

## Checklist

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸
